common-utils/src/common/deployment/__init__.py

```diff
@@ -356,125 +378,270 @@ class FlowSpec(BaseModel):
+        return final_task_def_arn
 
-    def _create_ecs_task_block(self):
-        """This method will create the ECS Task block as supported by common-utils."""
-        slugified_flow_name = standard_slugify(self.flow.name)
-        self._slugified_flow_name = slugified_flow_name
-        task_name = f"{slugified_flow_name}-{DEPLOYMENT_TYPE}"
-        block_name = f"{slugified_flow_name}-{ENVIRONMENT_TYPE}-{DEPLOYMENT_TYPE}"
-        account_id = get_aws_account_id()
+    def _create_ecs_task_block(
+        self, account_id: str, ecs_client: object, slugified_flow_name: str
+    ) -> str:
+        """This method will create the ECS Task block as supported by common-utils.
+
+        Args:
+            account_id (str): AWS account id
+            ecs_client (object): valid ecs client
+            slugified_flow_name (str): slugified flow name for naming
+            of task definition and block
+
+        Returns:
+            str: block name for use in deployment
+        """
+        # we only need prefect_aws when deploying the Block
+        from prefect_aws.ecs import ECSTask
+
+        # create Prefect ECSTask Block
+        # we are using a predefined task definition
+        # also turing off any additional handling of task definition in the ECSTask methods
+        block_name = f"{slugified_flow_name}-{DEPLOYMENT_TYPE}"
         ecs_block = ECSTask(
             name=block_name,
-            family=task_name,
-            image=f"{account_id}.dkr.ecr.{AWS_REGION}.amazonaws.com/data-flows-prefect-envs:{self.docker_env}",
-            cpu="256",
-            memory="512",
-            env={"PREFECT_DISABLE_FLOW_SPEC": "true"},
-            stream_output=True,
-            configure_cloudwatch_logs=True,
-            cluster=f"prefect-v2-agent-{ENVIRONMENT_TYPE}-{DEPLOYMENT_TYPE}",
-            execution_role_arn=f"arn:aws:iam::{account_id}:role/data-flows-prefect-{DEPLOYMENT_TYPE}-exec-role",
-            task_role_arn=f"arn:aws:iam::{account_id}:role/data-flows-prefect-{DEPLOYMENT_TYPE}-task-role",
-            task_definition={
-                "containerDefinitions": [
-                    {
-                        "secrets": [
-                            {
-                                "name": i.envar_name,
-                                "valueFrom": f"arn:aws:secretsmanager:{AWS_REGION}:{account_id}:secret:{i.secret_name}",
-                            }
-                            for i in self.secrets
-                        ],
-                    }
-                ],
-                "ephemeralStorage": {"sizeInGiB": self.ephemeral_storage_gb},
-            },
+            task_definition_arn=self._handle_task_definition(
+                account_id, ecs_client, slugified_flow_name
+            ),
+            cluster=f"prefect-v2-agent-{DEPLOYMENT_TYPE}",
+            launch_type="FARGATE",
+            allow_task_definition_registration=False,
```

Comment:
This method is much simpler and only imports the ECStask when needed.  This makes it so that local development does not need access to the prefect-aws library.